### ISSUE TITLE
Improve vehicle door choice

### ISSUE THREAD

**Issue by mcmonkey4eva** [mcmonkey4eva/GTAVRemultiplied#18]
We need vehicle door choice to feel more natural.

Perhaps tapping the button should enter driver door whenever it's available, and holding it allows alternate door choice.

We need to disable vehicle-auto-enter when holding it if we do that though.

Not sure how to design this.

**Comment by BlackCoyote** [comment]
if the situation prefers the player to be a passenger (as in, the driver seat is taken by a player or important NPC), tapping the button will make you enter the nearest free seat available, and holding it will make you try to carjack it

if the situation prefers the player to be a driver (empty car, random unimportant npc), tapping it will make you take the driver seat, and carjack if applicable.  I don't see it necessary for the player to have an immediate way into the vehicle as a passenger here. Any method for doing that would be scripted by the users of the mod

**Comment by mcmonkey4eva** [comment]
I agree with all but the last - I think having way to be a passenger ad hoc would be nice just to have the control available. Holding the button is unused in this place, so we might as well add "passenger riding with NPCs" as a server-permitted feature. (IE, the server has a config option to enable/disable this)

**Comment by BlackCoyote** [comment]
I wouldn't object, But I'm a little worried about that the controls are in that case entire opposites, holding means carjacking in case A, but passenger in B, and tapping means passenger in A, but carjacking in B

**Comment by BlackCoyote** [comment]
perhaps that is the most intuitive way, but on paper it does look like it might be a little bit confusing

**Comment by mcmonkey4eva** [comment]
Yeah it does sound a bit odd when you put it that way... let's try it and see how it goes, maybe?

**Comment by BlackCoyote** [comment]
that'll be the best way to find out!

**Comment by mcmonkey4eva** [comment]
Due to some oddities in GTAV updates, it no longer functions to have a player enter a specific seat via the AI Tasking engine... meaning I had to swap that for "teleport the player to the correct seat" for the moment... :/
